### DISCUSSION TITLE
Berry - adding a native lib

### DISCUSSION THREAD

**Discussion by btsimonh** [arendst/Tasmota#18849] [Q&A]
Taking the webclient lib as an example, I could see how to create the bones of a future webcam lib.

But  something eludes me:

```
#include "be_fixed_be_class_webclient.h"

void be_load_webclient_lib(bvm *vm) {
```

where is `be_fixed_be_class_webclient.h`, and where is be_load_webclient_lib called?
And does the processing to create the class get called every build? (I assume there is some script to process the commented berry class?)

I guess I'm asking for a short idiots guide to creating a native berry lib; what files are needed and how do they get included?

**Comment by s-hadinger** [comment]
You should start reading here: https://github.com/berry-lang/berry/wiki/Chapter-9

You have also the option of using berry_mapping (if the calls are not too often since parameters need to be parsed at each call): https://github.com/berry-lang/berry_mapping
